kornia/augmentation/_2d/mix/cutmix.py

```diff
@@ -133,3 +133,124 @@ def apply_transform(  # type: ignore
+                   probabilities batch-wisely.
+        num_mix (int): cut mix times. Default is 1.
+        beta (float or torch.Tensor, optional): hyperparameter for generating cut size from beta distribution.
+            Beta cannot be set to 0 after torch 1.8.0. If None, it will be set to 1.
+        cut_size ((float, float) or torch.Tensor, optional): controlling the minimum and maximum cut ratio from [0, 1].
+            If None, it will be set to [0, 1], which means no restriction.
+        same_on_batch (bool): apply the same transformation across the batch.
+            This flag will not maintain permutation order. Default: False.
+        keepdim (bool): whether to keep the output shape the same as input (True) or broadcast it
+                        to the batch form (False). Default: False
+
+    Inputs:
+        - Input image tensors, shape of :math:`(B, C, H, W)`.
+        - Raw labels, shape of :math:`(B)`.
+
+    Returns:
+        Tuple[torch.Tensor, torch.Tensor]:
+        - Adjusted image, shape of :math:`(B, C, H, W)`.
+        - Raw labels, permuted labels and lambdas for each mix, shape of :math:`(B, num_mix, 3)`.
+
+    Note:
+        This implementation would randomly cutmix images in a batch. Ideally, the larger batch size would be preferred.
+
+    Examples:
+        >>> rng = torch.manual_seed(3)
+        >>> input = torch.rand(2, 1, 3, 3)
+        >>> input[0] = torch.ones((1, 3, 3))
+        >>> label = torch.tensor([0, 1])
+        >>> cutmix = RandomCutMix()
+        >>> cutmix(input, label)
+        (tensor([[[[0.8879, 0.4510, 1.0000],
+                  [0.1498, 0.4015, 1.0000],
+                  [1.0000, 1.0000, 1.0000]]],
+        <BLANKLINE>
+        <BLANKLINE>
+                [[[1.0000, 1.0000, 0.7995],
+                  [1.0000, 1.0000, 0.0542],
+                  [0.4594, 0.1756, 0.9492]]]]), tensor([[[0.0000, 1.0000, 0.4444],
+                 [1.0000, 0.0000, 0.4444]]]))
+    """
+
+    def __init__(
+        self,
+        height: Optional[int] = None,
+        width: Optional[int] = None,
+        num_mix: int = 1,
+        cut_size: Optional[Union[torch.Tensor, Tuple[float, float]]] = None,
+        beta: Optional[Union[torch.Tensor, float]] = None,
+        same_on_batch: bool = False,
+        p: float = 1.0,
+        keepdim: bool = False,
+    ) -> None:
+        super().__init__(p=1.0, p_batch=p, same_on_batch=same_on_batch, keepdim=keepdim)
+        if height is not None or width is not None:
+            warnings.warn(
+                "height and width can be inferred automatically now. "
+                "The height and width arguments will be removed finally.",
+                category=DeprecationWarning,
+            )
+        self._param_generator = cast(rg.CutmixGenerator, rg.CutmixGenerator(cut_size, beta, num_mix, p=p))
+
+    def transform_labels(
+        self, label: torch.Tensor, params: Dict[str, torch.Tensor]
```

Comment:
```suggestion
        self, label: Tensor, params: Dict[str, Tensor]
```
Now you can do `from kornia.core import Tensor`